iGCS/Models/Craft/GCSCraftNotifications.m

```diff
@@ -24,4 +25,18 @@ + (void)didNavModeChangeFromLastHeartbeat:(mavlink_heartbeat_t) lastHeartbeat
     });
 }
 
++ (void)didMavModeChangeFromLastHeartbeat:(mavlink_heartbeat_t) lastHeartbeat
+                          andNewHeartbeat:(mavlink_heartbeat_t) newHeartbeat {
+
+    if (lastHeartbeat.base_mode != 0 &&
+        (lastHeartbeat.base_mode & MAV_MODE_FLAG_SAFETY_ARMED) == (newHeartbeat.base_mode & MAV_MODE_FLAG_SAFETY_ARMED)) {
+        return;
+    }
+
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [[NSNotificationCenter defaultCenter]postNotificationName:GCSCraftNotificationsCraftMavModeDidChange
```

Comment:
But the armed status and the custom mode status change is handled by separate events that are kicked off by entirely different properties of the heartbeat struct and it works as expected now (I think) what am I missing? Just poor method names are confusing us both or something else?